### PR TITLE
The Shore can redirect to shop.php

### PR DESCRIPTION
**Pull request by Rinn**

### CODE REVIEW

File: src/net/sourceforge/kolmafia/request/AdventureRequest.java [hunk 1]
```diff
@@ -1065,6 +1065,12 @@ public static final void handleServerRedirect(String redirectLocation) {
       return;
     }
 
+    if (redirectLocation.startsWith("shop.php")) {
+      // The Shore Inc. can redirect to the gift shop.
+      AdventureRequest.ZONE_UNLOCK.run();
+      return;
+    }
+
```
Comment (BadHorseMonkey) [inline src/net/sourceforge/kolmafia/request/AdventureRequest.java:1073]: I wish we could handle this (and the three identical code blocks above it) with something like a switch.  Are we far enough along on the Java version to get a better switch statement?

**Review by BadHorseMonkey — COMMENTED** [review]
Is there a unit test that could be written for this function?

File: src/net/sourceforge/kolmafia/request/AdventureRequest.java (hunk 1 above)
Comment (Rinn) [reply src/net/sourceforge/kolmafia/request/AdventureRequest.java:1073]: > Is there a unit test that could be written for this function?

I'm not writing a unit test.

Comment (gausie) [reply src/net/sourceforge/kolmafia/request/AdventureRequest.java:1073]: I have tried, there's no good test here for now.

**Review by gausie — APPROVED** [review]

**Pull request merged**